syft/generic/string.py

```diff
@@ -0,0 +1,301 @@
+            if isinstance(arg, self.__class__):
+                new_args.append(arg.core)
+            else:
+                new_args.append(arg)
+
+        return new_args
+
+    def method_hook(cls, attr: str):
+        """
+           This method creates and returns a function
+           that will be hooked to the wrapper class
+           being built by the current meta class.
+           This method will take the name 'attr' in the
+           wrapper class. It wraps the corresponding
+           method with the same name of the underlying
+           built-in type wrapped by the wrapper class.
+           
+           Args:
+               cls: A meta class object representing the current class
+               attr: A 'str' object representing the attribute name
+                     that is being hooked.
+
+           Return:
+              A Callable that will be used as the attribute 'attr'
+              of the new wrapping class.
+        """
+
+        def wrapper(self, *args, **kwargs):
+
+            args = cls._args_adaptor(self, args)
+
+            # Call the method of the core builtin type
+            native_output = getattr(self.core, attr)(*args, **kwargs)
+
+            # Some return types should be wrapped using an
+            # object of PySyft built-in-wrapper classes.
+            # For instance, if 'foo' is an object of type
+            # 'String' which wraps 'str'. calling foo.upper()
+            # should also be of type 'String' not 'str'.
+            # However, the return value of foo.__str__ should
+            # be of type 'str'.
+            output = cls._wrap_return_value(self, attr, native_output)
+
+            return output
+
+        return wrapper
+
+    def _wrap_return_value(self, attr: str, value: object):
+
+        # The outputs of the following attributed won't
+        # be wrapped
+        ignored_attr = set(["__str__", "__repr__", "__format__"])
+
+        if isinstance(value, str) and attr not in ignored_attr:
+            # TODO: change the __init__ call here
+            return String(object=value, owner=self.owner)
+
+        return value
+
+
+class String(AbstractObject, metaclass=BuiltinTypeWrapper, wrapped_type=str):
+    def __init__(
+        self,
+        object: object = None,
+        encoding: str = None,
+        errors: str = None,
+        id: int = None,
+        owner: BaseWorker = None,
+        tags: List[str] = None,
+        description: str = None,
+    ):
+        # get the specified kwargs for creating the base 'str'
+        # class
+
+        self.encoding = encoding
+        self.errors = errors
+
+        str_kwargs = {}
+
+        if object:
+            str_kwargs["object"] = object
+
+        if encoding:
+            str_kwargs["encoding"] = encoding
+
+        if errors:
+            str_kwargs["errors"] = errors
+
+        # Create a str instance as the 'child' attribute
+        self.core = str(**str_kwargs)
```

Comment:
If this is relevant, it might be more appropriate to call it `.child` to be consistent with the tensors api

Reply:
@LaRiffle Done